can/interfaces/robotell.py

```diff
@@ -0,0 +1,404 @@
+    _CAN_INIT_FLASH_ID = 0x01FFFEFF  # Restore factory settings
+    _CAN_READ_SERIAL1 = 0x01FFFFF0  # Read first part of device serial number
+    _CAN_READ_SERIAL2 = 0x01FFFFF1  # Read first part of device serial number
+    _MAX_CAN_BAUD = 1000000  # Maximum supported CAN baud rate
+    _FILTER_ID_MASK = 0x0000000F  # Filter ID mask
+    _CAN_FILTER_EXTENDED = 0x40000000  # Enable mask
+    _CAN_FILTER_ENABLE = 0x80000000  # Enable filter
+
+    _CAN_STANDARD_FMT = 0  # Standard message ID
+    _CAN_EXTENDED_FMT = 1  # 29 Bit extended format ID
+    _CAN_DATA_FRAME = 0  # Send data frame
+    _CAN_REMOTE_FRAME = 1  # Request remote frame
+
+    def __init__(
+        self, channel, ttyBaudrate=115200, bitrate=None, rtscts=False, **kwargs
+    ):
+        """
+        :param str channel:
+            port of underlying serial or usb device (e.g. /dev/ttyUSB0, COM8, ...)
+            Must not be empty.
+        :param int ttyBaudrate:
+            baudrate of underlying serial or usb device
+        :param int bitrate:
+            CAN Bitrate in bit/s. Value is stored in the adapter and will be used as default if no bitrate is specified
+        :param float poll_interval:
+            Poll interval in seconds when reading messages
+        :param bool rtscts:
+            turn hardware handshake (RTS/CTS) on and off
+        """
+
+        if not channel:  # if None or empty
+            raise TypeError("Must specify a serial port.")
+        if "@" in channel:
+            (channel, ttyBaudrate) = channel.split("@")
+        self.serialPortOrig = serial.serial_for_url(
+            channel, baudrate=ttyBaudrate, rtscts=rtscts
+        )
+
+        ## Disable flushing queued config ACKs on lookup channel (for unit tests)
+        self._loopback_test = channel == "loop://"
+
+        self._rxbuffer = bytearray()  # raw bytes from the serial port
+        self._rxmsg = []  # extracted CAN messages waiting to be read
+        self._configmsg = []  # extracted config channel messages
+
+        self._writeconfig(self._CAN_RESET_ID, 0)  # Not sure if this is really necessary
+
+        if bitrate is not None:
+            self.set_bitrate(bitrate)
+
+        self.channel_info = "Robotell USB-CAN s/n %s on %s" % (
+            self.get_serial_number(1),
+            channel,
+        )
+        logger.info("Using device: {}".format(self.channel_info))
+
+        super().__init__(
+            channel, ttyBaudrate=115200, bitrate=None, rtscts=False, **kwargs
+        )
+
+    def set_bitrate(self, bitrate):
+        """
+        :raise ValueError: if *bitrate* is greater than 1000000
+        :param int bitrate:
+            Bitrate in bit/s
+        """
+        if bitrate <= self._MAX_CAN_BAUD:
+            self._writeconfig(self._CAN_BAUD_ID, bitrate)
+        else:
+            raise ValueError(
+                "Invalid bitrate, must be less than " + str(self._MAX_CAN_BAUD)
+            )
+
+    def _set_auto_retransmit(self, retrans_flag):
```

Comment:
Any methods that you want to be public should not be prefixed with an underscore.

Reply:
Cleaned up, thanks.